Source/WebKit/UIProcess/API/glib/WebKitWebExtension.cpp

```diff
@@ -0,0 +1,1156 @@
+     * 
+     * Whether the #WebKitWebExtension has background content that stays in memory as long as the extension is loaded.
+     * See webkit_web_extension_get_has_persistent_background_content() for more details.
+     * 
+     * Since: 2.48
+     */
+    properties[PROP_HAS_PERSISTENT_BACKGROUND_CONTENT] =
+        g_param_spec_boolean(
+            "has-persistent-background-content",
+            nullptr, nullptr,
+            FALSE,
+            WEBKIT_PARAM_READABLE);
+
+    /**
+     * WebKitWebExtension:has-injected-content:
+     * 
+     * Whether the #WebKitWebExtension has script or stylesheet content that can be injected into webpages.
+     * See webkit_web_extension_get_has_injected_content() for more details.
+     * 
+     * Since: 2.48
+     */
+    properties[PROP_HAS_INJECTED_CONTENT] =
+        g_param_spec_boolean(
+            "has-injected-content",
+            nullptr, nullptr,
+            FALSE,
+            WEBKIT_PARAM_READABLE);
+
+    /**
+     * WebKitWebExtension:has-options-page:
+     * 
+     * Whether the #WebKitWebExtension has an options page.
+     * See webkit_web_extension_get_has_options_page() for more details.
+     * 
+     * Since: 2.48
+     */
+    properties[PROP_HAS_OPTIONS_PAGE] =
+        g_param_spec_boolean(
+            "has-options-page",
+            nullptr, nullptr,
+            FALSE,
+            WEBKIT_PARAM_READABLE);
+
+    /**
+     * WebKitWebExtension:has-override-new-tab-page:
+     * 
+     * Whether the #WebKitWebExtension provides an alternative to the default new tab page.
+     * See webkit_web_extension_get_has_override_new_tab_page() for more details.
+     * 
+     * Since: 2.48
+     */
+    properties[PROP_HAS_OVERRIDE_NEW_TAB_PAGE] =
+        g_param_spec_boolean(
+            "has-override-new-tab-page",
+            nullptr, nullptr,
+            FALSE,
+            WEBKIT_PARAM_READABLE);
+
+    /**
+     * WebKitWebExtension:has-commands:
+     * 
+     * Whether the #WebKitWebExtension includes commands that users can invoke.
+     * See webkit_web_extension_get_has_commands() for more details.
+     * 
+     * Since: 2.48
+     */
+    properties[PROP_HAS_COMMANDS] =
+        g_param_spec_boolean(
+            "has-commands",
+            nullptr, nullptr,
+            FALSE,
+            WEBKIT_PARAM_READABLE);
+
+    /**
+     * WebKitWebExtension:has-content-modification-rules:
+     * 
+     * Whether the #WebKitWebExtension includes rules used for content modification or blocking.
+     * See webkit_web_extension_get_content_modification_rules() for more details.
+     * 
+     * Since: 2.48
+     */
+    properties[PROP_HAS_CONTENT_MODIFICATION_RULES] =
+        g_param_spec_boolean(
+            "has-content-modification-rules",
+            nullptr, nullptr,
+            FALSE,
+            WEBKIT_PARAM_READABLE);
+
+    g_object_class_install_properties(objectClass, N_PROPERTIES, properties);
```

Comment:
...and then here we use the `std::array` accessors to get the pointer and number of elements:
```suggestion
    g_object_class_install_properties(objectClass, properties.size(), properties.data());
```